ext/cl_khr_suggested_local_work_size.asciidoc

```diff
@@ -0,0 +1,95 @@
+    cl_uint work_dim,
+    const size_t *global_work_offset,
+    const size_t *global_work_size,
+    size_t *suggested_local_work_size);
+----
+
+The returned suggested local work size is expected to match the local work size that would be chosen if the specified kernel object, with the same kernel arguments, were enqueued into the specified command queue with the specified global work size, specified global work offset, and with a `NULL` local work size.
+
+* _command_queue_ specifies the command queue and device for the query.
+* _kernel_ specifies the kernel object and kernel arguments for the query.
+The OpenCL context associated with _kernel_ and _command_queue_ must the same.
+* _work_dim_ specifies the number of work dimensions in the input global work offset and global work size, and the output suggested local work size.
+* _global_work_offset_ can be used to specify an array of at least _work_dim_ global ID offset values for the query.
+This is optional and may be `NULL` to indicate there is no global ID offset.
+* _global_work_size_ is an array of at least _work_dim_ values describing the global work size for the query.
+* _suggested_local_work_size_ is an output array of at least _work_dim_ values that will contain the result of the query.
+
+*clGetKernelSuggestedLocalWorkSizeKHR* returns `CL_SUCCESS` if the query executed successfully.
+Otherwise, it returns one of the following errors:
+
+* `CL_INVALID_COMMAND_QUEUE` if _command_queue_ is not a valid host command queue.
+* `CL_INVALID_KERNEL` if _kernel_ is not a valid kernel object.
+* `CL_INVALID_CONTEXT` if the context associated with _kernel_ is not the same as the context associated with _command_queue_.
+* `CL_INVALID_PROGRAM_EXECUTABLE` if there is no successfully built program executable available for _kernel_ for the device associated with _command_queue_.
+* `CL_INVALID_KERNEL_ARGS` if all argument values for _kernel_ have not been set.
+* `CL_MISALIGNED_SUB_BUFFER_OFFSET` if a sub-buffer object is set as an argument to _kernel_ and the offset specified when the sub-buffer object was created is not aligned to `CL_DEVICE_MEM_BASE_ADDR_ALIGN` for the device associated with _command_queue_.
+* `CL_INVALID_IMAGE_SIZE` if an image object is set as an argument to _kernel_ and the image dimensions are not supported by device associated with _command_queue_.
+* `CL_IMAGE_FORMAT_NOT_SUPPORTED` if an image object is set as an argument to _kernel_ and the image format is not supported by the device associated with _command_queue_.
+* `CL_INVALID_OPERATION` if an SVM pointer is set as an argument to _kernel_ and the device associated with _command_queue_ does not support SVM or the required SVM capabilities for the SVM pointer.
+* `CL_INVALID_WORK_DIMENSION` if _work_dim_ is not a valid value (i.e. a value between 1 and `CL_DEVICE_MAX_WORK_ITEM_DIMENSIONS`).
+* `CL_INVALID_GLOBAL_WORK_SIZE` if _global_work_size_ is NULL or if any of the values specified in _global_work_size_ are 0.
```

Comment:
I can't quite decide if this is sensible for OpenCL 2.1+ where this would be a legal call to `clEnqueueNDRangeKernel`, but returning values of `0` because of it seems dubious.

This is in contradiction with the "These error conditions are consistent with error conditions for clEnqueueNDRangeKernel." info box below, for OpenCL 2.1+.

The extension is explicitly "requires OpenCL 1.0" though, and this is too technical a thought for final editorial pass.  If I can't figure it out I'll file an issue.

Reply:
To be clear: This comment does not block merging, I just wrote down my thoughts.

Reply:
This was already discussed and resolved, I'm just late to the party.

```
//. What should the behavior be for a global work size equal to zero?
//+
//--
//*RESOLVED*: This is an error, both for a `NULL` global work size and when the global work size is equal to zero.
//The error code is `CL_INVALID_GLOBAL_WORK_SIZE`, consistent with *clEnqueueNDRangeKernel*.
```